### PR TITLE
command line args support

### PR DESCRIPTION
**Pull request by clemensv**
This PR consolidates the existing evaluation of environment variables into main() and introduces parameters into the utility functions where the environment is currently referenced directly.

The main() function uses yargs to parse parameters off the command line, with default values taken from the existing environment parameters. The app should thus run as before when using the .env file or explicitly set environment variables.

when using "npm start", the parameters must be passed by two (!) sets of double-dashes, e.g.

`npm start -- -- npm run start -- -- --twitter-handles clemensv --bluesky-username clemens-archive.vasters.com --archive-folder m:\twitter --bluesky-password ********** --min-date 2015-08-21`

That's a "feature" of npm, not of the script. 

This is in prep for bundling the app into a self-contained executable.

### CODE REVIEW

**Comment by marcomaroni-github** [comment]
Great work @clemensv, updated the readme accordingly? So when we get back to work we have an updated readme. Thanks a lot!

**Comment by clemensv** [comment]
Sure. I'll update the README.md in next couple of hours.

**Comment by clemensv** [comment]
@marcomaroni-github  README has been added. Please review and try it out before merging.

**Comment by clemensv** [comment]
> @clemensv I have tried this command line
> 
> `npm run start -- -- --archive-folder C:/Temp/twitter-archive-2024-04-08 --bluesky-username bsky-bot-test.bsky.social --bluesky-password 123 --twitter-handles marcomaroni --min-date '2023-08-27T03:12:49.393Z' --max-date '2023-08-28T03:12:49.393Z' --simulate`
> 

Does `node app.js  --archive-folder C:/Temp/twitter-archive-2024-04-08 --bluesky-username bsky-bot-test.bsky.social --bluesky-password 123 --twitter-handles marcomaroni --min-date '2023-08-27T03:12:49.393Z' --max-date '2023-08-28T03:12:49.393Z' --simulate` work for you?

I'm on Node 22

**Comment by marcomaroni-github** [comment]
> Does `node app.js --archive-folder C:/Temp/twitter-archive-2024-04-08 --bluesky-username bsky-bot-test.bsky.social --bluesky-password 123 --twitter-handles marcomaroni --min-date '2023-08-27T03:12:49.393Z' --max-date '2023-08-28T03:12:49.393Z' --simulate` work for you?

Yes, calling node directly it works, and I'm in node 20.12.1

**Comment by clemensv** [comment]
Then it's some change in behavior in npm, I assume

File: README.md [hunk 1]
```diff
@@ -103,14 +103,15 @@ Create an .env file in the project folder where you set the following variables
 - `BLUESKY_USERNAME` = username into which you want to import the tweets (e.g. "test.bsky.social")
 - `BLUESKY_PASSWORD` = account password created via App Password (eg. "pwd123")
 - `ARCHIVE_FOLDER` = full path to the folder containing the X/Twitter archive (e.g. "C:/Temp/twitter-archive")
-- `PAST_HANDLES` - one or more x/twitter handles without @, comma separated (e.g. 'marcomaroni,user'). Corresponds to the `--twitter-handles` argument.
+- `TWITER_HANDLES` - one or more x/twitter handles without @, comma separated (e.g. 'marcomaroni,user'). Corresponds to the `--twitter-handles` argument.
```
Comment (clemensv) [inline README.md]: Should be TWITTER

File: app.ts [hunk 2]
```diff
@@ -476,7 +473,7 @@ async function main() {
         .option('twitter-handles', {
             type: 'array',
             description: 'Twitter handles to import',
-            default: process.env.PAST_HANDLES?.split(','),
+            default: process.env.TWITER_HANDLES?.split(','),
```
Comment (clemensv) [inline app.ts]: Should be TWITTER

**Comment by clemensv** [comment]
Are you going to merge it?

**Comment by ricounet67** [comment]
I have the same issue (missing required arguments) and on Sequoia, with npm@10.9.1 and node.js v23.2.0
I tried with the command line and with a .env file, same result

**Pull request merged**